package cameraMsgs;

option java_package = "com.camera";

message cameraStatus {
	optional int64 time = 1;
	optional systemStatus status = 2 [default = UNKNOWN];
	optional bool laserOn = 3;
	optional string text = 4;
}

enum systemStatus {
	DOWN = 0;
	READY = 1;
	OPERATIONAL = 2;
	ERROR = 3;
	FAILED = 4;
	UNKNOWN = 5;
}

message track {
	optional sint32 x = 1;
	optional sint32 y = 2;
}

message cameraTracks {
	optional int64 time = 1;
	optional systemStatus status = 2 [default = UNKNOWN];
	
	repeated track target = 3;

	repeated track laser = 4;
	optional bool laserOn = 5;
}

message cameraImage {
	optional int64 time = 1;
	optional sint32 channels = 2;
	optional sint32 sizeX = 3;
	optional sint32 sizeY = 4;
	optional bytes imageData = 5;
}

